templates/install/tricks.md

```diff
@@ -0,0 +1,160 @@
+
+## Shared Mathlib
+
+If you start many projects which all use the latest stable version of mathlib
+by default each project will download its own clone.
+If you, for example, have little disk space available,
+it might be worth setting them up using one centralised copy of mathlib instead.
+
+*Note*: This means additional effort when upgrading the mathlib version,
+as you need to update all your projects at once.
+
+*Note*: Whenever this tutorial mentions the `lakefile.lean`, you should make the mentioned
+modifications to your `lakefile.toml` if you have this instead. Every Lean project
+has exactly one of these two configuration files.
+
+Here is the current best practice to achieve this.
+
+1) First, clone a version of mathlib somewhere on your computer:
+   ```bash
+   git clone --branch v4.10.0 git@github.com:leanprover-community/mathlib4.git
+   ```
+   Note that `v4.10.0` is the tag of the latest release, you can look at [mathlib's tags](https://github.com/leanprover-community/mathlib4/tags) to find out which is the most recent release version.
+
+   The above line assumes you have set up git using an SSH key.
+   If you have not set this up correctly, you may want to
+   use `git clone --branch v4.10.0 https://github.com/leanprover-community/mathlib4.git` instead.
+
+2) Go inside your mathlib and download the current cache:
+   ```bash
+   cd mathlib
+   lake exe cache get
+   ```
+3) If you ever want to **update** your global mathlib, come back to the mathlib directory and call
+   ```bash
+   git checkout v4.11.0
+   lake exe cache get
+   ```
+   with the [version](https://github.com/leanprover-community/mathlib4/tags) you'd like to update to.
+4) If you don't already have a Lean project, create it.
+   ```bash
+   lake new MyProject math.lean
+   cd MyProject
+   ```
+5) In the project `MyProject` you need to modify the following things:
+   * Make sure `lean-toolchain` contains the string `leanprover/lean4:v4.10.0` with the same version your shared mathlib is at.
+   * In `lakefile.lean`, replace the line `require "leanprover-community" / "mathlib"` with
+     ```
+     require mathlib from ".." / "relative" / "path" / "to" / "mathlib4"
+     ```
+   * Now inside `MyProject` you need to clean up lake:
+     ```bash
+     rm -rf .lake # because `lake clean` does not remove `.lake/packages/mathlib` which might have been downloaded by `lake new`.
+     lake clean # or potentially `lake update -R mathlib` instead
+     ```
+     *(note: it looks like a bug that with a simple `lake clean`, there might still be a folder `.lake/packages/mathlib` floating around from before you changed the `lakefile.lean`. However, deleting `.lake/` is a reasonably safe action as it only contains build artifacts that are fully recovered by the next `lake` call.)*
```

Comment:
```suggestion
     ```bash
     lake update -R mathlib # use the new path for mathlib
     rm -rf .lake/packages/mathlib # delete the previous local clone of mathlib
     ```
```

`lake update -R mathlib` should be enough to switch to the relative mathlib. If not, that is a bug. Also, `lake clean` does not delete Lake packages, it merely removes build outputs.